### DISCUSSION TITLE
How to install missing PHP extensions under Arch Linux?

### DISCUSSION THREAD

**Discussion by Scratch7951** [datenstrom/community#656] [Ask a question] [answered]
Yellow requires php extensions `curl exif gd mbstring zip`. I'm running Arch Linux with PHP 8.0.6 and both `php-exif` and `php-gd` are missing from the AUR. I can't run Yellow without them, but I cannot install them either.

Any advice on what I can do? I'm stumped.

Thanks.

**Comment by schulle4u** [comment, marked as answer]
As this is no yellow-specific problem I suggest to check the [troubleshoting section for php](https://wiki.archlinux.org/title/PHP#Troubleshooting) in the Arch wiki. I'm not an Arch user, but the suggestions sound helpful to me.

**Reply by Scratch7951** [reply]
Thanks for the link to the Troubleshooting section of the Archwiki. I'd read it several times but missed [this crucial example](https://wiki.archlinux.org/title/PHP#gd) (e.g. installing `gd`)

The solution is to edit `/etc/php/php.ini` and uncomment the lines for the extensions you want to use. This is how 'gd' and 'exif' functionality can be added to the system's php environment.

As a user with no `php` experience, it's confusing that some php extensions must be installed from the AUR but others are already installed but inactive by default. For `gd` and `exif` it's a matter of activating them by editing `php.ini`; for e.g. `zip` you have to install from the AUR.

Anyway, Yellow is now working fine. Thank you very much for the tip @schulle4u .

**Reply by schulle4u** [reply]
And that's why I prefer Debian-based distros. They often are easier to install and maintain for everyday usage, although Archlinux has it's own advantages and someday I might have a closer look. Happy coding! 😀

**Reply by markseuffert** [reply]
I removed the exif dependency, to make the installation on Linux a little bit easier.
